rest-api-spec/src/main/resources/rest-api-spec/test/pit/10_basic.yml

```diff
@@ -138,5 +138,7 @@
   - match: {pits.0.successful: true }
 
   - do:
-      catch: missing
-      delete_all_pits: { }
+      delete_all_pits: {}
```

Comment:
Does this change the behavior of deleting a single PIT or a list of PITs? Should there be a test for those cases as well?

Reply:
@andrross I don't think it will result in any change in the behaviour of deleting a single PIT or a list of PITs using `DELETE localhost:9200/_search/point_in_time/` API.

### Request Delete PIT with empty array results in validation exception.
DELETE localhost:9200/_search/point_in_time/
Request Body
```
{
    "pit_id": [
        ]
}
```
Response
```
{
	"error": {
		"root_cause": [
			{
				"type": "action_request_validation_exception",
				"reason": "Validation Failed: 1: no pit ids specified;"
			}
		],
		"type": "action_request_validation_exception",
		"reason": "Validation Failed: 1: no pit ids specified;"
	},
	"status": 400
}
```

### Request with incorrect pits results in failure with failure message associated with every pit id
DELETE localhost:9200/_search/point_in_time/
Request Body
```
{
    "pit_id": [
        "o463QQEPbXktaW5kZXgtMDAwMDAxFkhGN09fMVlPUkVPLXh6MUExZ1hpaEEAFjBGbmVEZHdGU1EtaFhhUFc4ZkR5cWcAAAAAAAAAAAEWaXBPNVJtZEhTZDZXTWFFR05waXdWZwEWSEY3T18xWU9SRU8teHoxQTFnWGloQQAA",
        "o463QQEPbXktaW5kZXgtMDAwMDAxFkhGN09fMVlPUkVPLXh6MUExZ1hpaEEAFjBGbmVEZHdGU1EtaFhhUFc4ZkR5cWcAAAAAAAAAAAIWaXBPNVJtZEhTZDZXTWFFR05waXdWZwEWSEY3T18xWU9SRU8teHoxQTFnWGloQQAA"
    ]
}
```
Response
```
{
	"pits": [
		{
			"successful": false,
			"pit_id": "o463QQEPbXktaW5kZXgtMDAwMDAxFkhGN09fMVlPUkVPLXh6MUExZ1hpaEEAFjBGbmVEZHdGU1EtaFhhUFc4ZkR5cWcAAAAAAAAAAAEWaXBPNVJtZEhTZDZXTWFFR05waXdWZwEWSEY3T18xWU9SRU8teHoxQTFnWGloQQAA"
		},
		{
			"successful": false,
			"pit_id": "o463QQEPbXktaW5kZXgtMDAwMDAxFkhGN09fMVlPUkVPLXh6MUExZ1hpaEEAFjBGbmVEZHdGU1EtaFhhUFc4ZkR5cWcAAAAAAAAAAAIWaXBPNVJtZEhTZDZXTWFFR05waXdWZwEWSEY3T18xWU9SRU8teHoxQTFnWGloQQAA"
		}
	]
}
```

Reply:
`Should there be a test for those cases as well?`
Few tests cases already exist for the DELETE PIT(s). But I couldn't find test case for the use case where empty pit array is passed in the body.
```
DELETE localhost:9200/_search/point_in_time/
Request Body:
{
    "pit_id": []
}
```
I believe this should be tracked through a separate issue.